cfme/tests/cli/test_appliance_update.py

```diff
@@ -0,0 +1,78 @@
+import tempfile
+
+import pytest
+
+from cfme.test_framework.sprout.client import SproutClient, SproutException
+from scripts.repo_gen import process_url, build_file
+from utils.version import Version
+from utils.log import logger
+from utils.conf import cfme_data
+from utils.appliance import current_appliance
+from utils import os
+
+versions = []
+
+
+def pytest_generate_tests(metafunc):
+    """The following lines generate appliance versions based from the current build.
+    Appliance version is split and minor_build is picked out for generating each version
+    and appending it to the empty versions list"""
+
+    version = current_appliance.version
+    split_ver = str(version).split(".")
+    minor_build = split_ver[2]
+
+    for i in range(int(minor_build) - 1, -1, -1):
+        versions.append("{}.{}.{}".format(split_ver[0], split_ver[1], i))
+
+
+@pytest.yield_fixture(scope="function")
+def appliance_preupdate(old_version, appliance):
+
+    series = appliance.version.series()
+    update_url = "update_url_{}".format(series[0] + series[2])
+
+    """Requests appliance from sprout based on old_versions, edits partitions and adds
+    repo file for update"""
+
+    usable = []
+    sp = SproutClient.from_config()
+    available_versions = set(sp.call_method('available_cfme_versions'))
+    for ver in versions:
+        for a in available_versions:
+            if a.startswith(ver):
+                usable.append(a)
+    usable = sorted([Version(i) for i in usable], reverse=True)
```

Comment:
Version is needed to sort the list in the correct order, as a standard sort messes up alph/beta build versions.